wallet-new/src/Cardano/Wallet/Kernel/DB/Sqlite.hs

```diff
@@ -129,84 +132,132 @@ instance Beamable (PrimaryKey TxMetaT)
 {--
 
 * Table 2: ~tx_meta_inputs~
-** Primary Index: ~tx_meta_input_address~
-** Secondary Indexes: ~tx_meta_id~
-
-| tx_meta_input_address | tx_meta_coin | tx_meta_id |
-|-----------------------+--------------+------------|
-| Core.Address          | Core.Coin    | Txp.TxId  |
+** Primary key: ~input_id~, ~input_foreign_txid~, ~input_foreign_index~
 
-** Table 3: ~tx_meta_outputs~
-** Primary Index: ~tx_meta_output_address~
-** Secondary Indexes: ~tx_meta_id~
+input_id   | input_address | input_coin | input_foreign_txid | input_foreign_index
+-----------|---------------+------------+--------------------+--------------------
+Txp.TxId   | Core.Address  | Core.Coin  | Txp.TxId           | Word32
 
-| tx_meta_output_address | tx_meta_coin | tx_meta_id |
-|------------------------+--------------+------------|
-| Core.Address           | Core.Coin    | Txp.TxId  |
 
 --}
 
-data TxCoinDistributionTableT f = TxCoinDistributionTable {
-      _txCoinDistributionTableAddress :: Beam.Columnar f Core.Address
-    , _txCoinDistributionTableCoin    :: Beam.Columnar f Core.Coin
-    , _txCoinDistributionTxId         :: Beam.PrimaryKey TxMetaT f
-    } deriving Generic
-
-type TxCoinDistributionTable = TxCoinDistributionTableT Identity
-
-instance Beamable TxCoinDistributionTableT
-
 -- | The inputs' table.
-newtype TxInputT f = TxInput  {
-  _getTxInput  :: (TxCoinDistributionTableT f)
-  } deriving Generic
+data TxInputT f = TxInputT {
+      _inputTableTxId         :: Beam.PrimaryKey TxMetaT f
+    , _inputTableAddress      :: Beam.Columnar f Core.Address
+    , _inputTableCoin         :: Beam.Columnar f Core.Coin
+    , _inputTableForeignTxid  :: Beam.Columnar f Txp.TxId
+    , _inputTableForeignIndex :: Beam.Columnar f Word32
+    } deriving Generic
```

Comment:
Could I be annoying here and ask that you change the order? The order of a UTxO entry basically _everywhere_ is `((txid, ix), (addr, coin))`. I think it would help if we follow the same structure here.